src/main/java/io/cdap/plugin/couchbase/sink/RecordToJsonDocumentTransformer.java

```diff
@@ -0,0 +1,243 @@
+                          schema.getEnumValues()));
+        }
+        return enumValue;
+      default:
+        throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'", fieldName,
+                                                          fieldLogicalType.name().toLowerCase()));
+    }
+  }
+
+  /**
+   * Get UTC zoned date and time represented by the specified timestamp in microseconds.
+   *
+   * @param ts timestamp in microseconds
+   * @return UTC {@link ZonedDateTime} corresponding to the specified timestamp
+   */
+  private ZonedDateTime extractUTCDateTime(long ts) {
+    ZoneId zoneId = ZoneId.ofOffset("UTC", ZoneOffset.UTC);
+    long mod = TimeUnit.MICROSECONDS.convert(1, TimeUnit.SECONDS);
+    int fraction = (int) (ts % mod);
+    long tsInSeconds = TimeUnit.MICROSECONDS.toSeconds(ts);
+    // create an Instant with time in seconds and fraction which will be stored as nano seconds.
+    Instant instant = Instant.ofEpochSecond(tsInSeconds, TimeUnit.MICROSECONDS.toNanos(fraction));
+    return ZonedDateTime.ofInstant(instant, zoneId);
+  }
+
+  private JsonObject extractMap(String fieldName, Map<String, ?> map, Schema schema) {
+    if (map == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+    Map.Entry<Schema, Schema> mapSchema = Objects.requireNonNull(schema.getMapSchema());
+
+    JsonObject jsonObject = JsonObject.create();
+    Schema valueSchema = mapSchema.getValue();
+    Schema nonNullableSchema = mapSchema.getValue().isNullable() ? valueSchema.getNonNullable() : valueSchema;
+    map.forEach((k, v) -> jsonObject.put(k, extractValue(fieldName, v, nonNullableSchema)));
+
+    return jsonObject;
+  }
+
+  private JsonArray extractArray(String fieldName, Object value, Schema schema) {
+    if (value == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+
+    Schema componentSchema = schema.getComponentSchema();
+    Schema nonNullableSchema = componentSchema.isNullable() ? componentSchema.getNonNullable() : componentSchema;
+    JsonArray extracted = JsonArray.create();
+    if (value.getClass().isArray()) {
+      int length = Array.getLength(value);
+      for (int i = 0; i < length; i++) {
+        Object arrayElement = Array.get(value, i);
+        extracted.add(extractValue(fieldName, arrayElement, nonNullableSchema));
+      }
+      return extracted;
+    }
+    // An 'array' field can be a java.util.Collection or an array.
+    Collection values = (Collection) value;
+    for (Object obj : values) {
+      extracted.add(extractValue(fieldName, obj, nonNullableSchema));
+    }
+    return extracted;
+  }
+
+  private Object extractUnion(String fieldName, Object value, Schema schema) {
+    if (value == null) {
+      // Return 'null' value as it is
+      return null;
+    }
+    for (Schema s : schema.getUnionSchemas()) {
+      try {
+        return extractValue(fieldName, value, s);
+      } catch (Exception e) {
```

Comment:
too wide exception class

Reply:
Changed to `ClassCastException`. Thanks.